src/k8s_etcd.rs

```diff
@@ -176,6 +176,36 @@ impl InMemoryK8sEtcd {
         self.deleted_keys.lock().await.insert(key.to_string());
         Ok(())
     }
+
+    pub(crate) async fn update_member(&self, value: String) -> Result<()> {
+        let etcd_client = self.etcd_client.clone().context("etcd client not configured")?;
+
+        let members_list = etcd_client
+            .cluster_client()
+            .member_list()
+            .await
+            .context("listing etcd members list")?;
+
+        ensure!(
+            members_list.members().len() == 1,
+            "single-node must have exactly one etcd member, found {}",
+            members_list.members().len()
+        );
+
+        if let Some(member) = members_list.members().first() {
+            ensure!(
+                !etcd_client
+                    .cluster_client()
+                    .member_update(member.id(), vec![value.clone()])
+                    .await
+                    .context("updating etcd member")?
+                    .members()
+                    .is_empty(),
+                "no members in update response"
+            )
+        };
```

Comment:
With the above `ensure!` we can safely remove the `if` statement:

```suggestion
        ensure!(
            !etcd_client
                .cluster_client()
                .member_update(members[0].id(), vec![value.clone()])
                .await
                .context("updating etcd member")?
                .members()
                .is_empty(),
            "no members in update response"
        )
```

Reply:
But since every time we call `members()` it's an API call to etcd maybe it's best to save the members list in a variable. The full code should look like:

```rust
        let members = members_list.members();

        ensure!(
            members.len() == 1,
            "single-node must have exactly one etcd member, found {}",
            members.len()
        );

        ensure!(
            !etcd_client
                .cluster_client()
                .member_update(members[0].id(), vec![value.clone()])
                .await
                .context("updating etcd member")?
                .members()
                .is_empty(),
            "no members in update response"
        );
```